lib/puppet/defaults.rb

```diff
@@ -1517,8 +1517,7 @@ def self.default_diffargs
     },
     :lastrunreport =>  {
       :default  => "$statedir/last_run_report.yaml",
-      :type     => :file,
-      :mode     => "0640",
+      :type     => :path,
```

Comment:
It saves to multiple locations.

From my testing, a lastrunreport is never read by puppet - that's why #find was not updated.